logging/tests/unit/handlers/test_app_engine.py

```diff
@@ -12,46 +12,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+import os
 import unittest
 
 
 class TestAppEngineHandlerHandler(unittest.TestCase):
-    PROJECT = 'PROJECT'
+    from google.cloud.logging.resource import Resource
+
+    GAE_PROJECT_ENV = 'PROJECT'
+
+    GAE_RESOURCE = Resource(
+        type='gae_app',
+        labels={
+            'project_id': os.environ.get(GAE_PROJECT_ENV),
+            'module_id': os.environ.get('GAE_SERVICE'),
+            'version_id': os.environ.get('GAE_VERSION'),
+        },
+    )
 
     def _get_target_class(self):
         from google.cloud.logging.handlers.app_engine import AppEngineHandler
 
         return AppEngineHandler
 
     def _make_one(self, *args, **kw):
-        import tempfile
-
-        from google.cloud._testing import _Monkey
-        from google.cloud.logging.handlers import app_engine as _MUT
+        return self._get_target_class()(*args, **kw)
 
-        tmpdir = tempfile.mktemp()
-        with _Monkey(_MUT, _LOG_PATH_TEMPLATE=tmpdir):
-            return self._get_target_class()(*args, **kw)
+    def test_ctor(self):
+        client = _Client(self.GAE_PROJECT_ENV)
+        handler = self._make_one(client, transport=_Transport)
+        self.assertEqual(handler.client, client)
+        self.assertEqual(handler.resource, self.GAE_RESOURCE)
```

Comment:
Instead of reflecting the object, maybe just pluck out the labels and name:

```python
self.assertEqual(handler.resource.type, 'gae_app')
self.assertEqual(handler.labels['project_id'], ...)
...
```